app/services/waveform_service.rb

```diff
@@ -44,10 +44,18 @@ def get_normalized_peaks(uri)
   end
 
   def get_wave_io(uri)
-    headers = "-headers $'Referer: #{Rails.application.routes.url_helpers.root_url}\r\n'" if uri.starts_with? "http"
-    normalized_uri = uri.starts_with?("file") ? URI.unescape(uri) : uri
-    cmd = "#{Settings.ffmpeg.path} #{headers} -i '#{normalized_uri}' -f wav -ar 44100 - 2> /dev/null"
-    IO.popen(cmd)
+    # use array syntax for filenames with apostrophes; general command line injection
+    # add an optional header for when using http and not a file
+    cmd = [Settings.ffmpeg.path,
+           "-i",
+           uri.starts_with?("file") ? URI.unescape(uri) : uri, # normalize url 
+           '-f',
+           'wav',
+           '-ar',
+           '44100']
+    cmd << "-headers $'Referer: #{Rails.application.routes.url_helpers.root_url}\r\n'" if uri.starts_with? "http"
+    cmd << 'pipe:1'
```

Comment:
Is the Ruby fix to use an array of strings to properly escape the apostrophe and prevent command-line injection? Or is there a better way